base/rational.jl

```diff
@@ -533,3 +533,13 @@ function hash(x::Rational{<:BitInteger64}, h::UInt)
     h = hash_integer(num, h)
     return h
 end
+
+# This method is only needed for performance. Since `first(r)` and `last(r)` have the same
+# denominator (because their difference is an integer), `length(r)` can be calulated without
+# calling `gcd`.
+function length(r::AbstractUnitRange{T}) where T<:Rational
+    @_inline_meta
+    f = first(r)
+    l = last(r)
+    return div(l.num - f.num + f.den, f.den)
```

Comment:
```suggestion
    a = fld(l.num, l.den) - fld(f.num, f.den)
    return a + oneunit(a)
```
This change would reduce the risk of overflow, but would be slightly less performant (two integer divisions instead of one). Is it worth it?